benches/coprocessor_executors.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extern crate criterion;
+
+extern crate test_coprocessor;
+extern crate tikv;
+extern crate tipb;
+
+use criterion::{black_box, Bencher, Criterion};
+
+use tipb::executor::TableScan;
+
+use test_coprocessor::*;
+use tikv::coprocessor::codec::Datum;
+use tikv::coprocessor::dag::executor::{Executor, TableScanExecutor};
+
+fn bench_table_scan_primary_key(b: &mut Bencher) {
+    let id = ColumnBuilder::new()
+        .col_type(TYPE_LONG)
+        .primary_key(true)
+        .build();
+    let foo = ColumnBuilder::new().col_type(TYPE_LONG).build();
+    let table = TableBuilder::new()
+        .add_col(id.clone())
+        .add_col(foo.clone())
+        .build();
+
+    let mut store = SimpleBTreeMapStore::new();
+    for i in 0..10 {
+        store.begin();
+        store
+            .insert_into(&table)
+            .set(&id, Datum::I64(i))
+            .set(&foo, Datum::I64(0xDEADBEEF))
+            .execute();
+        store.commit();
+    }
+
+    let mut meta = TableScan::new();
+    meta.set_table_id(table.id);
+    meta.set_desc(false);
+    meta.mut_columns().push(id.get_column_info());
+
+    b.iter_with_setup(
+        || {
+            let mut executor = TableScanExecutor::new(
+                meta.clone(),
+                vec![table.get_select_range()],
+                store.clone().into_fixture_store(),
```

Comment:
Maybe we don't really need a `SimpleBTreeMap` and we can simply implement `into_fixture_store` for `MvccTransactionalStore`?

Reply:
Sounds great! I will have a try.